controllers/front/GuestTrackingController.php

```diff
@@ -81,12 +81,28 @@ public function postProcess()
             $customer = new Customer((int) $this->order->id_customer);
             $password = Tools::getValue('password');
 
-            if (strlen($password) < Validate::PASSWORD_LENGTH) {
+            if (empty($password)) {
+                $this->errors[] = $this->trans(
+                    'Enter a password to transform your guest account into a customer account.',
+                    [],
+                    'Shop.Forms.Help'
+                );
+            } elseif (strlen($password) < Validate::PASSWORD_LENGTH) {
                 $this->errors[] = $this->trans(
                     'Your password must be at least %min% characters long.',
                     ['%min%' => Validate::PASSWORD_LENGTH],
                     'Shop.Forms.Help'
                 );
+            // Prevent error
+            // A) either on page refresh
+            // B) if we already transformed him in other window or through backoffice
+            } elseif ($customer->is_guest == 0) {
+                $this->errors[] = $this->trans(
+                    'A customer account has already been created from this guest account. Please sign in.',
+                    [],
+                    'Shop.Notifications.Success'
```

Comment:
```suggestion
                    'Shop.Notifications.Error'
```

Sorry for the delay, I was out of the office last week. I just missed this domain. It's actually an error, not a success. :-)

Reply:
@Julievrz No worries. Good catch, completely forgot to check it :). I also fixed it in confirmation controller. 👍